SharedSource/UnitTests/FunctionalProgramming/EnumerableExtensionsTest.cs

```diff
@@ -182,14 +212,14 @@ public void CreateSequence_WhilePredicateEvaluatesTrue ()
       var third = new Element (3, second);
       var fourth = new Element (4, third);
 
-      IEnumerable<Element> actual = fourth.CreateSequence (e => e.Parent, e => e != first);
+      IEnumerable<Element?> actual = ((Element?)fourth).CreateSequence (e => e!.Parent, e => e != first);
       Assert.That (actual.ToArray (), Is.EqualTo (new[] { fourth, third, second }));
     }
 
     [Test]
     public void CreateSequence_WhilePredicateEvaluatesTrue_WithNull ()
     {
-      IEnumerable<Element> actual = ((Element)null).CreateSequence (e => e.Parent, e => e != null);
+      IEnumerable<Element> actual = ((Element?)null).CreateSequence (e => e!.Parent, e => e != null)!;

```

Comment:
Can't we write this as 
((Element)null!).CreateSequence  (e => e.Parent, e= e!= null)

Reply:
CreateSequence would then require `Func<T, T>` but `e => e.Parent` is `Func<T, T?>` and would again require a NFO